File: 0782-jewels-and-stones/0782-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int count = 0;
        Set<Character> freqs = new HashSet<>();

        for (char jewel : jewels.toCharArray()) freqs.add(jewel);

        for (char stone : stones.toCharArray())
            if (freqs.contains(stone)) count++;
        
        return count;
    }
}

File: 0782-jewels-and-stones/0782-jewels-and-stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        var count = 0
        val freqs: MutableSet<Char> = HashSet()

        for (jewel in jewels.toCharArray()) freqs.add(jewel)

        for (stone in stones.toCharArray())
            if (freqs.contains(stone)) count++
        
        return count
    }
}